ui/src/components/swap/Swap.tsx

```diff
@@ -0,0 +1,220 @@
+        give: { ...Object.fromEntries(gives), Fee: fee },
+        want: { ...Object.fromEntries(wants) },
+      };
+
+      makeOffer(
+        invitationSpec,
+        proposal,
+        offerArgs,
+        (update: { status: string; data?: unknown }) => {
+          if (update.status === 'seated') {
+            addNotification!({
+              text: `Offer Sent`,
+              status: 'success',
+            });
+          }
+          if (update.status === 'error') {
+            addNotification!({
+              text: `Offer error: ${update.data}`,
+              status: 'error',
+            });
+          }
+          if (update.status === 'accepted') {
+            addNotification!({
+              text: 'Offer accepted',
+              status: 'success',
+            });
+          }
+          if (update.status === 'refunded') {
+            addNotification!({
+              text: 'Offer refunded',
+              status: 'warning',
+            });
+          }
+        },
+      );
+    } catch (e) {
+      addNotification!({
+        text: `Offer error: ${e}`,
+        status: 'error',
+      });
+    }
+  };
+
+  const isButtonDisabled =
+    !makeOffer ||
+    !recipientAddr ||
+    !(myAmounts.length || recipientAmounts.length);
+
+  return (
+    <div className="items-top flex w-full flex-col justify-around lg:flex-row">
+      <div>
+        <h2 className="daisyui-card-title mb-2 w-full">Create Swap Offer</h2>
+        <div className="daisyui-card h-fit w-96 bg-base-300 px-4 py-4 shadow-xl">
+          <RecipientInput
+            address={recipientAddr}
+            onChange={addr => setRecipientAddr(addr)}
+            error={recipientError}
+          />
+          <div className="my-1">
+            <h2 className="mb-2 text-lg font-medium">Give</h2>
+            <ProposalAmountsBox
+              actionLabel="Add from Your Purse"
+              amounts={myAmounts}
+              purses={purses}
+              onChange={setMyAmounts}
+              warning={purses ? undefined : 'Wallet Not Connected'}
+            />
+          </div>
+          <div className="my-1">
+            <h2 className="mb-2 text-lg font-medium">Want</h2>
+            <ProposalAmountsBox
+              actionLabel="Add from Recipient Purse"
+              amounts={recipientAmounts}
+              onChange={setRecipientAmounts}
+              purses={recipientPurses}
+              warning={
+                recipientPurses.length ? undefined : 'Enter Recipient Address'
+              }
+            />
+          </div>
+          {fee && <FeeInfo fee={fee} />}
+          <button
+            onClick={sendOffer}
+            disabled={isButtonDisabled}
+            className="daisyui-btn daisyui-btn-primary mt-4 w-full self-center text-lg"
+          >
+            Send Offer
+          </button>
+        </div>
+      </div>
+      <div className="daisyui-divider lg:daisyui-divider-horizontal"></div>
+      <IncomingOffers />
```

Comment:
wondering if this component could be reused in our ui-kit. Seems it would be very useful for many dapps 🚀